src/table.rs

```diff
@@ -61,40 +62,351 @@ impl Table {
+            }
+        }
+    }
+
+    fn take_provided_action(&mut self, hand_action: HandAction, active_state: ActiveState) {
+        let difference = self.get_max_bet() - active_state.current_bet;
+        // Now check how to advance the hand
+        match hand_action {
+            HandAction::Fold => {
+                self.get_current_player().fold();
+            }
+            HandAction::Check => {
+                if difference == 0 {
+                    self.get_current_player().bet(0);
+                } else {
+                    self.get_current_player().fold();
+                }
+            }
+            HandAction::Call => {
+                self.pot += self.get_current_player().bet(difference);
+            }
+            HandAction::Raise(raise_amount) => {
+                self.pot += self.get_current_player().bet(difference + raise_amount);
+            }
+        }
+        // Update to point at the next player
+        self.update_current_player_index_to_next_active();
+    }
+
+    pub fn get_results(&self) -> String {
+        let mut players_copy = self.players.clone();
+        players_copy.sort_by(|a, b| b.cmp(a));
+        let mut rank = 1;
+        let mut result_string = format!("Results:\nRank: {}, Player: {}\n", rank, players_copy.get(0).unwrap());
+        for (i, player) in players_copy.iter().skip(1).enumerate() {
+            // The players didn't tie, so increase the rank
+            if player != players_copy.get(i).unwrap() {
+                rank = i + 2;
+            }
+            result_string += &format!("Rank: {rank}, Player: {player}\n");
+        }
+        result_string
+    }
 
+    /// Gets the current turn information
+    pub fn get_current_turn_information(&mut self) -> JsonValue {
+        self.get_state_string_for_player(self.players.get(self.current_player_index).unwrap().get_id())
+    }
+
+    /// Cleans all the cards from the table
+    pub fn clean_table(&mut self) {
+        self.flop = None;
+        self.turn = None;
+        self.river = None;
+        for player in &mut self.players {
+            player.fold();
+        }
+    }
+
+    pub fn is_game_over(&self) -> bool {
+        let alive_player_count = self.players.iter().map(|x| i8::from(x.is_alive())).reduce(|x, y| x + y).unwrap();
+        alive_player_count == 1
+    }
+
+    /// Deals cards to all players that are still alive,
+    /// marks any dead players with their death turn number,
+    /// moves the dealer chip,
+    /// update all 5 table cards,
     pub fn deal(&mut self) {
+        // If the game is over do not do anything
+        if self.is_game_over() {
+            return;
+        }
+        // Increment the hand number
+        self.hand_number += 1;
+        // Set the pot back to zero
+        self.pot = 0;
+        // Reset the state for a new round of betting
+        self.reset_state_for_new_round();
+        // Check all players for death
+        self.check_for_player_death();
```

Comment:
If ante increases, we'll want to make sure it happens before this check

Reply:
Adding an issue.